src/main/java/com/wavefront/opentracing/reporting/WavefrontSpanReporter.java

```diff
@@ -21,15 +27,27 @@
  *
  * @author Vikram Raman (vikram@wavefront.com)
  */
-public class WavefrontSpanReporter implements Reporter {
-  private static final Logger logger =
-      Logger.getLogger(WavefrontSpanReporter.class.getName());
+public class WavefrontSpanReporter implements Reporter, Runnable {
+  private static final Logger logger = Logger.getLogger(WavefrontSpanReporter.class.getName());
 
   private final WavefrontSender wavefrontSender;
   private final String source;
+  private final LinkedBlockingQueue<WavefrontSpan> spanBuffer;
+  private final Thread sendingThread;
+  private final Random random;
+  private final float logPercent;
+
+  private volatile WavefrontInternalReporter metricsReporter;
```

Comment:
I see that you did this because you did not want to change the signature of 
WavefrontSpanReporter
Can you add a comment as to why you are using a setter and why it is volatile?